ui/pages/send/gas-display/gas-display.js

```diff
@@ -0,0 +1,337 @@
+import { NETWORK_TO_NAME_MAP } from '../../../../shared/constants/network';
+import TransactionDetail from '../../../components/app/transaction-detail/transaction-detail.component';
+import ActionableMessage from '../../../components/ui/actionable-message/actionable-message';
+import DepositPopover from '../../../components/app/deposit-popover/deposit-popover';
+import { getProvider } from '../../../selectors';
+import { INSUFFICIENT_TOKENS_ERROR } from '../send.constants';
+
+const renderHeartBeatIfNotInTest = () =>
+  process.env.IN_TEST ? null : <LoadingHeartBeat />;
+export default function GasDisplay({
+  primaryTotalTextOverride,
+  secondaryTotalTextOverride,
+  hexTransactionTotal,
+  draftTransaction,
+  useNativeCurrencyAsPrimaryCurrency,
+  primaryTotalTextOverrideMaxAmount,
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+  isMainnet,
+  showLedgerSteps,
+  isBuyableChain,
+  hexMinimumTransactionFee,
+  hexMaximumTransactionFee,
+  nativeCurrency,
+  chainId,
+  showAccountDetails,
+  gasError,
+}) {
+  const t = useContext(I18nContext);
+  const { estimateUsed, supportsEIP1559V2 } = useGasFeeContext();
+  const [showDepositPopover, setShowDepositPopover] = useState(false);
+  const currentProvider = useSelector(getProvider);
+  const networkName = NETWORK_TO_NAME_MAP[chainId];
+  const isInsufficientTokenError =
+    draftTransaction?.amount.error === INSUFFICIENT_TOKENS_ERROR;
+  const renderTotalMaxAmount = () => {
+    if (
+      primaryTotalTextOverride === undefined &&
+      secondaryTotalTextOverride === undefined &&
+      draftTransaction.asset.type === 'NATIVE'
+    ) {
+      // Native Send
+      return (
+        <UserPreferencedCurrencyDisplay
+          type={PRIMARY}
+          key="total-max-amount"
+          value={addHexes(
+            draftTransaction.amount.value,
+            hexMaximumTransactionFee,
+          )}
+          hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+        />
+      );
+    }
+    // Token send
+    return useNativeCurrencyAsPrimaryCurrency
+      ? primaryTotalTextOverrideMaxAmount
+      : secondaryTotalTextOverride;
+  };
+  const renderTotalDetailTotal = () => {
+    if (
+      primaryTotalTextOverride === undefined &&
+      secondaryTotalTextOverride === undefined &&
+      draftTransaction.asset.type === 'NATIVE'
+    ) {
+      return (
+        <div className="confirm-page-container-content__total-value">
+          <LoadingHeartBeat
+            estimateUsed={draftTransaction?.userFeeLevel ?? estimateUsed}
+          />
+          <UserPreferencedCurrencyDisplay
+            type={PRIMARY}
+            key="total-detail-value"
+            value={hexTransactionTotal}
+            hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+          />
+        </div>
+      );
+    }
+    return useNativeCurrencyAsPrimaryCurrency
+      ? primaryTotalTextOverrideMaxAmount
+      : secondaryTotalTextOverride;
+  };
```

Comment:
I would prefer these to go inline in the props instead of as intermediary functions that are invoked.

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/30e2c90ae66cf2a12aeb11e5ea35daa7cfd60340).